src/Model/Resolver/Product/PriceRange.php

```diff
@@ -135,7 +137,53 @@ protected function formatPrice(float $regularPrice, float $finalPrice, StoreInte
                 'value' => $finalPrice,
                 'currency' => $store->getCurrentCurrencyCode()
             ],
-            'discount' => $this->discount->getDiscountByDifference($regularPrice, $finalPrice),
+            'discount' => $discount,
         ];
     }
+
+    /**
+     * Calculates correct discount amount
+     * - Bundle items can contain $regularPrice and $finalFrice from two different
+     * - product instances, thus we are intersted in BE set special price procentage.
+     *
+     * @param Product $product
+     * @param float $regularPrice
+     * @param float $finalPrice
+     * @return array
+     */
+    protected function calculateDiscount(Product $product, float $regularPrice, float $finalPrice) : array
+    {
+        if ($product->getTypeId() != 'bundle') {
+            return $this->discount->getDiscountByDifference($regularPrice, $finalPrice);
+        }
+
+        $specialPrice = $this->getSpecialProductPrice($product);
+        $percentOff = ($specialPrice < 0) ? 0 : 100 - $specialPrice;
+
+        return [
+            'amount_off' => $regularPrice * ($percentOff / 100),
+            'percent_off' => $percentOff
+        ];
+    }
+
+    /**
+     * Gets [active] special price value
+     *
+     * @param Product $product
+     * @return float
+     */
+    protected function getSpecialProductPrice(Product $product): float
+    {
+        $specialPrice = $product->getSpecialPrice();
+        if (!$specialPrice) {
+            return -1;
+        }
+
+        // Special price range
+        $from = strtotime($product->getSpecialFromDate());
+        $to = $product->getSpecialToDate() === null ? null : strtotime($product->getSpecialToDate());
+        $now = time();
+
+        return ($now >= $from && $now <= $to || $now >= $from && is_null($to)) ? (float)$specialPrice : -1;
```

Comment:
In some languages, without braces, the operator priority sequence does not work, IDK if PHP is one of them but please prefer having braces when working with logical operations. 
`($now >= $from && $now <= $to) || ($now >= $from && is_null($to))`

Reply:
Fixed. Added braces for logical operation